### PR TITLE
serverless subgraph checker

### PR DESCRIPTION
**Pull request by hoanganhsb**
https://startbahn.atlassian.net/browse/STARTRAIL-1174

I will add readme.md and test cases later

### CODE REVIEW

File: checker/src/functions/check-subgraph/handler.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+  CHECKER_RUNTIME_SUBGRAPH_URL: string
+  CHECKER_RUNTIME_ETHEREUM_PROVIDER_URL: string
+  CHECKER_RUNTIMETHRESHOLD_BLOCKS_LAGGING: string
+}
+
+// load from runtime environment variables
+const getConfig = (): ConfigType => {
+  return {
+    CHECKER_RUNTIME_SUBGRAPH_URL: process.env.CHECKER_RUNTIME_SUBGRAPH_URL!,
+    CHECKER_RUNTIME_ETHEREUM_PROVIDER_URL: process.env
```
Comment (chatch) [inline checker/src/functions/check-subgraph/handler.ts:15]: This breaks @typescript-eslint/no-non-null-assertion when I read the code in vscode. I think it's picking up the rules from subgraph main eslint config ..

**Review by chatch — COMMENTED** [review]
looks really good @hoanganhsb :+1:  nicely set up.

i've added some comments about some configuration setup points.

take a look at those and as you mention on the comments add a unit test and I'll review again after that.

re: public vs private comment in slack, yes I agree, sounds good to make it private.

**Comment by hoanganhsb** [comment]
@chatch updated

**Comment by chatch** [comment]
@anhnguyenhoang thank you Anh, all changes look good.

can i request one more which is to rename instances of `stag` to `stg` to be consistent with how the startrail api refers to it.

**Comment by hoanganhsb** [comment]
@chatch updated (https://console.cloud.google.com/functions/details/asia-northeast1/subgraph-stg-checkSubgraph?authuser=2&project=startrail-api-dev)

**Review by chatch — APPROVED** [review]
looks good, approved.

**Comment by chatch** [comment]
@wshino I've approved this PR however could you take a quick look over this from an infrastructure point of view? Thank you.

**Comment by chatch** [comment]
btw. i'm going to copy this setup for the signing server cloud function.  :+1:

**Comment by chatch** [comment]
@hoanganhsb I just copied the checker/ repo and I'm starting to edit files for my signing server function. I realised there are quite a lot of files to look through, especially configuration. Comparing with an [example serverless project](https://github.com/serverless/examples/tree/master/google-node-simple-http-endpoint) which is really mimimal. 

But the example doesn't have extra features you've added like eslint etc. but I wonder what do you think we'd lose by going to a really simple setup like this? maybe the ability to configure everything for each env. easily? 

I dont' think we need to change it on this PR but just curious about that. Going forward as we replicate the setup you've built here we might be able to streamline it a bit ..

**Comment by chatch** [comment]
@hoanganhsb I noticed a couple of points looking through the config that I didn't notice in the review.

1) There is an inconsistency with a variable named "..RUNTIME_THRESHOLD..". In one place it has an underscore and in another it does not. 

2) package.json stg has NODE_ENV=development but qa has NODE_ENV=production. I think let's make all of these production. All deployed versions to GCP should be the same.

**Comment by chatch** [comment]
ah actually i see now the template `google-nodejs-typescript` and you've based it on that right? yes that has nice extras over the super simple one.

**Comment by chatch** [comment]
@wshino i should have mentioned we have https://startbahn.atlassian.net/browse/STARTRAIL-1182 to set up deployments on circle ci.  but just good to look over this first setup before we get to that. btw. i just noticed there is a serverless orb which might be handy: https://github.com/CircleCI-Public/serverless-framework-orb

**Review by wshino — APPROVED** [review]
thanks @hoanganhsb, in the future, I will automate this deploy with circleci.

**Comment by hoanganhsb** [comment]
> ah actually i see now the template `google-nodejs-typescript` and you've based it on that right? yes that has nice extras over the super simple one.

Yes

**Comment by hoanganhsb** [comment]
> @hoanganhsb I noticed a couple of points looking through the config that I didn't notice in the review.
> 
> 1. There is an inconsistency with a variable named "..RUNTIME_THRESHOLD..". In one place it has an underscore and in another it does not.
> 2. package.json stg has NODE_ENV=development but qa has NODE_ENV=production. I think let's make all of these production. All deployed versions to GCP should be the same.

Ok. I will fix that today

**Comment by hoanganhsb** [comment]
> 1. RUNTIME_THRESHOLD

@chatch fixed

**Comment by hoanganhsb** [comment]
> @hoanganhsb I just copied the checker/ repo and I'm starting to edit files for my signing server function. I realised there are quite a lot of files to look through, especially configuration. Comparing with an [example serverless project](https://github.com/serverless/examples/tree/master/google-node-simple-http-endpoint) which is really mimimal.
> 
> But the example doesn't have extra features you've added like eslint etc. but I wonder what do you think we'd lose by going to a really simple setup like this? maybe the ability to configure everything for each env. easily?
> 
> I dont' think we need to change it on this PR but just curious about that. Going forward as we replicate the setup you've built here we might be able to streamline it a bit ..

@chatch I think so

**Comment by hoanganhsb** [comment]
@chatch updated

**Pull request merged**